PyInstaller/compat.py

```diff
@@ -863,3 +863,33 @@ def check_requirements():
     # Fail hard if Python does not have minimum required version
     if sys.version_info < (3, 3) and sys.version_info[:2] != (2, 7):
         raise SystemExit('PyInstaller requires at least Python 2.7 or 3.3+.')
+
+if is_py2:
+    class suppress(object):
+        """Context manager to suppress specified exceptions
+        After the exception is suppressed, execution proceeds with the next
+        statement following the with statement.
+             with suppress(FileNotFoundError):
+                 os.remove(somefile)
+             # Execution still resumes here if the file was already removed
+        """
+
+        def __init__(self, *exceptions):
+            self._exceptions = exceptions
+
+        def __enter__(self):
+            pass
+
+        def __exit__(self, exctype, excinst, exctb):
+            # Unlike isinstance and issubclass, CPython exception handling
+            # currently only looks at the concrete type hierarchy (ignoring
+            # the instance and subclass checking hooks). While Guido considers
+            # that a bug rather than a feature, it's a fairly hard one to fix
+            # due to various internal implementation details. suppress provides
+            # the simpler issubclass based semantics, rather than trying to
+            # exactly reproduce the limitations of the CPython interpreter.
+            #
+            # See http://bugs.python.org/issue12029 for more details
+            return exctype is not None and issubclass(exctype, self._exceptions)
+else:
+    from contextlib import suppress
```

Comment:
This is used in the test suite only, right? Then it should go into e.g. PyInstaller/utils/tests.py or a new module PyInstaller/utils/tests_compat.py

Reply:
It's currently used in the test suite, but the idea is to include it for future use elsewhere in PyInstaller.